sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/statsbeat/_customer_statsbeat.py

```diff
@@ -221,3 +230,56 @@ def _get_retry_reason(self, retry_code: RetryCodeType, exception_message: Option
             RetryCode.UNKNOWN: "unknown_reason",
         }
         return retry_code_reasons.get(retry_code, "unknown_reason")
+
+# Global customer statsbeat singleton
+_CUSTOMER_STATSBEAT_METRICS = None
+_CUSTOMER_STATSBEAT_LOCK = threading.Lock()
+
+
+# pylint: disable=global-statement
+# pylint: disable=protected-access
+def collect_customer_statsbeat(exporter):
+    global _CUSTOMER_STATSBEAT_METRICS
+    # Only start customer statsbeat if did not exist before
+    if _CUSTOMER_STATSBEAT_METRICS is None:
+        with _CUSTOMER_STATSBEAT_LOCK:
+            # Double-check inside the lock to avoid race conditions
+            if _CUSTOMER_STATSBEAT_METRICS is None:
+                class StatsbeatOptions:
+                    def __init__(self, connection_string):
+                        self.connection_string = connection_string
+
+                connection_string = (
+                    f"InstrumentationKey={exporter._instrumentation_key};"
+                    f"IngestionEndpoint={exporter._endpoint}"
+                )
+                statsbeat_options = StatsbeatOptions(connection_string)
+                _CUSTOMER_STATSBEAT_METRICS = CustomerStatsbeatMetrics(statsbeat_options)
+
+    exporter._customer_statsbeat_metrics = _CUSTOMER_STATSBEAT_METRICS
+    if hasattr(exporter, 'storage') and exporter.storage:
+        exporter.storage._customer_statsbeat_metrics = _CUSTOMER_STATSBEAT_METRICS
+
+_CUSTOMER_STATSBEAT_STATE = {
+    "SHUTDOWN": False,
+}
+_CUSTOMER_STATSBEAT_STATE_LOCK = threading.Lock()
+
+def shutdown_customer_statsbeat_metrics() -> None:
+    global _CUSTOMER_STATSBEAT_METRICS
+    shutdown_success = False
+    if _CUSTOMER_STATSBEAT_METRICS is not None:
+        with _CUSTOMER_STATSBEAT_LOCK:
+            try:
+                if _CUSTOMER_STATSBEAT_METRICS._meter_provider is not None:
+                    _CUSTOMER_STATSBEAT_METRICS._meter_provider.shutdown()
+                    _CUSTOMER_STATSBEAT_METRICS = None
+                    shutdown_success = True
+            except:  # pylint: disable=bare-except
+                pass
+        if shutdown_success:
+            with _CUSTOMER_STATSBEAT_STATE_LOCK:
+                _CUSTOMER_STATSBEAT_STATE["SHUTDOWN"] = True
+
+def get_customer_statsbeat_shutdown():
```

Comment:
For next pr: Could probably put this in `state.py` similar to regular statsbeat.